DFT+U and linear response U

avadean opened the pull request:
I have added some documentation about the DFT+U method and the calculation of the U parameter via a linear response method. This latter method has been implemented but not yet pulled into the CASTEP repository. It is planned to be accessible via a development code on its first merge into the code, and the documentation reflects this.

For DFT+U, the new pages are:
- Background and theory
- Input

For the linear response method to calculate U, the new pages are:
- Background and theory
- Input
- Output
- Warnings and errors

The method depends on a parameter named alpha, hence the file names.

Commit:
Added overview of DFT+U and the linear response method to calculate U. Next is a tutorial on the newly implemented linear response method to calculate U.

Commit:
Some corrections and improvements to the DFT+U and linear response overviews.

Commit:
Added discussion on input, output and errors of the Linear Response Calculation of U method

avadean committed:
Merge branch 'develop' into hubbard_alpha

jryates commented:
This is great. I think we should hold off to merge it until the code is itself merged.
We also need to make it clear to users which functionality is in which version of the code. So we need a note (like with the NEB docs) to say what is in castep now, and what will be in, say, v25.

avadean commented:
Hi! I know this has been a long time. Apologies for that.

I am going to make some updates to this now that the Koopmans code has been merged. Therefore converted to draft.

Changes will include proper cell/param inputs (no longer devel code), and some corrections/improvements to my explanation of the Cococcioni & de Gironcoli method.

avadean committed:
Merge branch 'develop' into hubbard_alpha

avadean committed:
Corrected specification of units in hubbard alpha block.

Updated the method of input (no longer devel code).

Outlined different ways to specify hubbard alpha values.

Added mentions of Koopmans' compliance in theory sections.

Added references for combinations of DFT+U and computed U values.

Greatly improved the outline of the Cococcioni and de Gironcoli method.

Updated charge neutrality equations.

Added discussion of intercept error checking messages.

avadean committed:
Changed name of linear response calculation of U from alpha to koopmans

avadean committed:
Added notes in relavent sections to outline that a user must be using CASTEP version 26 or later to access the Koopman;s compliance functionality to determine the Hubbard U parameter
